developing/configuration/README.md

```diff
@@ -0,0 +1,13 @@
+# Configuration
+
+Projects should adhere to the following guidelines for managing configuration:
+
+1. Configuration values should be provided to programs via environment variables.
+2. Programs should validate configuration at boot time.
```

Comment:
is it maybe worth clarifying `program boot time` to distinguish from "OS boot time"?

Reply:
Isn't this "run time" (ie, when the program is run)? Or am I misunderstanding a nuance?

Reply:
How about "programs should validate configuration when they are run"? I was trying to communicate that it should occur immediately when the program starts and not at some point later on.